### PR TITLE
Consolidate host directives

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: project/build.properties [hunk 1]
```diff
@@ -1 +1 @@
-sbt.version=1.0.4
+sbt.version=1.1.0-RC2
```
Comment (jrudolph) [inline project/build.properties]: Sneaky, do we really want that here? ;)

File: docs/src/main/paradox/java/http/routing-dsl/directives/alphabetically.md [hunk 2]
```diff
@@ -37,7 +37,7 @@
...
 |@ref[extractMaterializer](basic-directives/extractMaterializer.md)                                                   | Extracts the @unidoc[Materializer] from the @unidoc[RequestContext]                                                                                                                                                               |
-|@ref[extractHost](host-directives/extractHostName.md)                                                                | Extracts the hostname part of the Host request header value                                                                                                                                                         |
+|@ref[extractHostName](host-directives/extractHost.md)                                                                | Extracts the hostname part of the Host request header value                                                                                                                                                         |
```
Comment (jrudolph) [inline docs/src/main/paradox/java/http/routing-dsl/directives/alphabetically.md]: There doesn't seem to be any `extractHostName` directive (any more?). I guess we can use `extractHost` now also for Java.

File: docs/src/main/paradox/scala/http/routing-dsl/directives/host-directives/index.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ contained in incoming requests as well as extracting its value for usage in inne
 
 @@@ index
 
-* [extractHost](extractHost.md)
 * [host](host.md)
+* @scala[[extractHost](extractHost.md)]@java[[extractHostName](extractHost.md)]
```
Comment (jrudolph) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/host-directives/index.md]: Why change the order?

File: project/build.properties (hunk 1 above)
Comment (ktoso) [reply project/build.properties]: I think this has a chance to fix the javac NPE issue we've seen. Perhaps as different mini PR though?

Comment (raboof) [reply project/build.properties]: Oops, that wasn't intentional here :)

File: docs/src/main/paradox/java/http/routing-dsl/directives/alphabetically.md (hunk 2 above)
Comment (raboof) [reply docs/src/main/paradox/java/http/routing-dsl/directives/alphabetically.md]: ah even better, I didn't realize ;)

**Pull request merged**